sdk/personalizer/Azure.AI.Personalizer/src/Models/PersonalizerClient.cs

```diff
@@ -20,11 +20,16 @@ public class PersonalizerClient
         private readonly HttpPipeline _pipeline;
         private readonly bool _isLocalInference;
         private string stringEndpoint;
-        private string apiKey;
+        private AzureKeyCredential _azureKeyCredential;
+        private TokenCredential _tokenCredential;
+        private int liveModelRefreshTimeInMinutes = 15;
+        private string _credentialType;
```

Comment:
Better to define `_credentialType` as an enum than a string. String operations are error prone, expensive, and are not IDE friendly.  With that said, is `_credentialType` even necessary? Why not the following?

```csharp
if (_tokenCredential != null) {
// do work
} else if (_azureKeyCredential != null) {
// do work
}
```